python/cudf/cudf/core/indexing.py

```diff
@@ -113,7 +113,16 @@ def _loc_to_iloc(self, arg):
             return indices_from_labels(self._sr, arg)
 
         if isinstance(
-            arg, (list, np.ndarray, pd.Series, range, Index, DeviceNDArray)
+            arg,
+            (
+                list,
+                np.ndarray,
+                pd.Series,
+                range,
+                Index,
+                DeviceNDArray,
+                cupy.ndarray,
```

Comment:
Lets use `__cuda_array_interface__` instead of hardcoding `cupy.ndarray` if possible. We should similarly use `__array_interface__` for numpy-like objects.

Reply:
Can't use `__array_interface__` as numpy scalars also gets considered.

Reply:
We could move the `is_scalar` check above here?